src/tags/jsonset.js

```diff
@@ -30,7 +30,7 @@ module.exports =
 
             let varname = undefined;
             const deserializedTagArray = await bu.deserializeTagArray(value);
-            if (deserializedTagArray)
+            if (deserializedTagArray && !deserializedTagArray.n)
```

Comment:
Should be `if (deserializedTagArray && deserializedTagArray.n)` as you are checking for a value in the variable `deserializedTagArray` but also checking that this object has a property `n` different from a falsy value.

Reply:
I actually thought the same, but if the input array is of the format {"v": ["1","2","3"], "n": "~array"}, it will return the same input object but just parsed. 

You want to make sure that the input array isn't a 'variable array (to keep the reference to the variable, and allow modifications to the array similar to how JS works with references)', and that's why I added `!deserializedTagArray.n`, as that will return a falsy if it doesn't have a name (and thus isn't an array variable).

Like I mentioned on Discord, if you want to pass on an array variable without keeping the reference to the variable, you'd use `{slice;{get;~array};0}.

Reply:
So what has been decided?

Reply:
Keeping the if statement as is? I'm not entirely sure what's the remaining issue. Your request says it should be acceptable for arrays to be passed without name (not a variable), and that's kind off the only thing that this if statement handles. It only accepts arrays passed without name.